src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets

```diff
@@ -369,7 +379,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     ============================================================
   -->
   <Target Name="_GetRestoreProjectStyle"
-    DependsOnTargets="_GetProjectJsonPath"
+    DependsOnTargets="_GetProjectJsonPath;CollectPackageReferences"
```

Comment:
To determine if a project is packages.config vs PackageReference restore checks for PackageReference items, this should read them in the same way as when it actually pulls in the references.

For command line restore there is little difference between NETCore SDK projects and legacy csproj, in VS I suppose this target would not get called unfortunately.